src/components/encounter/encounter-form-manager.ts

```diff
@@ -114,23 +114,36 @@ export class EncounterFormManager {
     patient: fhir.Patient,
     currentPatientPrograms: Array<PatientProgram>,
   ): Array<PatientProgramPayload> {
-    const programFields = fields.filter((field) => field.type === 'programState' && hasSubmission(field));
-    return programFields.map((field) => {
+    const programStateFields = fields.filter((field) => field.type === 'programState' && hasSubmission(field));
+    const programMap = new Map<string, PatientProgramPayload>();
+    programStateFields.forEach((field) => {
       const programUuid = field.questionOptions.programUuid;
+      const newState = field.meta.submission.newValue;
       const existingProgramEnrollment = currentPatientPrograms.find((program) => program.program.uuid === programUuid);
+
       if (existingProgramEnrollment) {
-        return {
-          uuid: existingProgramEnrollment.uuid,
-          states: [field.meta.submission.newValue],
-        };
+        if (programMap.has(programUuid)) {
+          programMap.get(programUuid).states.push(newState);
+        } else {
+          programMap.set(programUuid, {
```

Comment:
Thanks @samuelmale for this, quick one, do you only set what is needed for the update payload?

Reply:
Yes that's correct. I only set the necessary props for the update.